go-controller/pkg/ovn/namespace.go

```diff
@@ -425,3 +427,31 @@ func (oc *Controller) deleteNamespaceLocked(ns string) *namespaceInfo {
 
 	return nsInfo
 }
+
+func (oc *Controller) cleanUpNsinfo(nsInfo *namespaceInfo) {
+
+	klog.V(5).Infof("Deleting Namespaces's NetworkPolicy entities")
+
+	for _, np := range nsInfo.networkPolicies {
+		delete(nsInfo.networkPolicies, np.name)
```

Comment:
Can we not share code with deleteNetworkPolicy()?

Reply:
So `deleteNetworkPolicy()` is called with a `*knet.NetworkPolicy` which we don't have access to within the `namespace`  object (all we have is the Namespace stored *namespacePolicy), and `deleteNetworkPolicy()` works great except for this edge case where the namespace object manages to delete before the policy elements.  So therefore It would be harder to share and refactor the the code in `deleteNetworkPolicy` rather than add a temporary addendum to `namespace.go` I believe.  Ultimately I don't think we should have network policies be dependent on namespaces in OVN since we receive the events for deletion of both entities asynchronously.

Reply:
@dcbw Unless you have a suggestion that I'm not understanding :)

Reply:
they both use the namespacePolicy struct that is part of nsInfo...  you could define `cleanUpNamespacePolicy(np *namespacePolicy)` in policy.go and call it from both `cleanUpNsinfo()` and `deleteNamespace()`

Reply:
ok I see, thanks updating the patch now